lib/OpenQA/Shared/Plugin/Gru.pm

```diff
@@ -40,18 +40,31 @@ sub register_tasks ($self) {
       );
 }
 
+# allow the continuously polled stats to be available on an
+# unauthenticated route to prevent recurring broken requests to the login
+# provider if not logged in
+#
+# hard/impossible to mock due to name collision of "remove" method on
+# Test::MockObject, hence marking as
```

Comment:
You could just use a sub package like in the worker tests instead of `Test::MockObject`. Or you just use a real app by creating one via `Mojolicious->new` (which is not that expensive).

Reply:
True but would it be better than this approach? I am not sure and I would prefer to use proper mocking modules where possible.

Reply:
Not mocking things always increases the changes that we actually test what we need to test. I'd normally only mock things if not mocking has a too big performance impact. I guess in this case it doesn't really matter, though.

I would not consider `Test::MockModule` more "proper" than just creating sub packages. If you make the sub packages inherit from `Mojo::Base` there's very little boilerplate code and code is actually easier to read. Just because someone wrote a Perl module it doesn't mean it is automatically the better approach to use it - and the API of `Test::MockModule` is not *that* intuitive.